First push minimap

geuks opened the pull request:
Adding minimap

geuks committed:
First push minimap

Adding minimap

CJ-SPT commented on UI/ModdedMapScreen.cs:
```diff
@@ -241,7 +271,12 @@ private void Update()
         internal void OnMapScreenShow()
         {
-            _peekComponent?.EndPeek();
+            PeekComponent?.EndPeek();
```
This should be an explicit null check, unity overrides this and can lead to issues. Same with every other occurrence inside of a monobehavior

geuks committed:
polish

Adding security to component when its called
Adding default value for settings and category for minimap (f12)
Fix auto map level
Fix zoom in/out when press key
Revert key for ZoomMapOut (mistake)

mpstark commented:
I'm not currently playing the game and don't have a way to test this PR. I'm glad that you've done this, but at a bit of an impasse without a currently active maintainer.

geuks commented:
> I'm not currently playing the game and don't have a way to test this PR. I'm glad that you've done this, but at a bit of an impasse without a currently active maintainer.

Thanks for your response. I understand the situation with no active maintainer at the moment. I initially made this for myself, but I decided to contribute because I think others would appreciate having this feature. No pressure to review or merge it right now.
